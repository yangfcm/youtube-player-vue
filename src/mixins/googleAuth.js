import { mapActions } from "vuex";
const googleAuth = {
  created() {
    if (!window.gapi) return;
    window.gapi.load("client:auth2", () => {
      window.gapi.client
        .init({
          clientId: process.env.VUE_APP_CLIENT_ID,
          scope:
            "email https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/youtube https://www.googleapis.com/auth/youtube.force-ssl",
        })
        .then(() => {
          this.auth = window.gapi.auth2.getAuthInstance();
          this.onAuthChange(this.auth.isSignedIn.get());
          this.auth.isSignedIn.listen(this.onAuthChange);
        });
    });
  },

  methods: {
    ...mapActions({
      signIn: "signIn",
      signOut: "signOut",
    }),
    onAuthChange(isSignedIn) {
      if (isSignedIn) {
        this.signIn(this.auth.currentUser.get()); // signIn action
      } else {
        this.signOut(); // signOut action
      }
    },
    handleGoogleSignout() {
      this.auth.signOut(); // signOut provided by Google Auth API
    },
    handleGoogleSignin() {
      this.auth.signIn(); // signIn provided by Google Auth API
    },
  },
};

export default googleAuth;
