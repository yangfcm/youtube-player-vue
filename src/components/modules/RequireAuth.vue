<template>
  <div class="ui red message">
    <app-blank></app-blank>
    <div class="ui header aligned center">You have not signed in</div>
    <app-blank></app-blank>
    <div class="ui header aligned center">
      <button class="ui youtube button negative basic" @click="handleSignIn">
        <i class="youtube icon"></i>
        Sign in
      </button>
    </div>
    <app-blank></app-blank>
  </div>
</template>

<script>
import googleAuth from "../../mixins/googleAuth";
import Blank from "../common/Blank";
export default {
  components: {
    appBlank: Blank,
  },
  mixins: [googleAuth],
  methods: {
    handleSignIn() {
      this.handleGoogleSignin();
    },
  },
};
</script>

<style scoped>
</style>