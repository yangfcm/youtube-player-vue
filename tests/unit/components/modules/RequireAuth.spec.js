import { shallowMount } from "@vue/test-utils";
import RequireAuth from "@/components/modules/RequireAuth";
import googleAuth from "@/mixins/googleAuth";

jest.mock("@/mixins/googleAuth");

describe("Test RequireAuth component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(RequireAuth);
  });
  afterEach(() => {
    wrapper.destroy();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render require signin information", () => {
    expect(wrapper.text()).toContain("You have not signed in");
  });

  it("should render Sign in button", () => {
    expect(wrapper.find("button").text()).toContain("Sign in");
  });

  it("clicking sign in button should call sign in method", () => {
    wrapper.find("button").trigger("click");
    expect(googleAuth.methods.handleGoogleSignin).toHaveBeenCalled();
  });
});
