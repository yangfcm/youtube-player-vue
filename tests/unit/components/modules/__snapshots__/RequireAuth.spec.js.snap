// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RequireAuth component should match snapshot 1`] = `
<div class="ui red message">
  <app-blank-stub></app-blank-stub>
  <div class="ui header aligned center">You have not signed in</div>
  <app-blank-stub></app-blank-stub>
  <div class="ui header aligned center"><button class="ui youtube button negative basic"><i class="youtube icon"></i>
      Sign in
    </button></div>
  <app-blank-stub></app-blank-stub>
</div>
`;
